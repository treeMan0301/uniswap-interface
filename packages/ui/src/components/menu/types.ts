import { BaseSyntheticEvent } from 'react'
import { GeneratedIcon, IconProps } from 'ui/src/components/factories/createIcon'
import { TextProps } from 'ui/src/components/text'
import { TouchableAreaProps } from 'ui/src/components/touchable'

export type MenuContentItem = {
  label: string
  onPress: (e: BaseSyntheticEvent) => void
  textProps?: TextProps
  Icon?: GeneratedIcon | ((props: IconProps) => JSX.Element)
  iconProps?: IconProps
  destructive?: boolean
} & TouchableAreaProps
