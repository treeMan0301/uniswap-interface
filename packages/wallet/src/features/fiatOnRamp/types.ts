import { FiatPurchaseTransactionInfo, TransactionDetails } from 'wallet/src/features/transactions/types'

export type FiatOnRampTransactionDetails = TransactionDetails & {
  typeInfo: FiatPurchaseTransactionInfo
}

/** @ref https://dashboard.moonpay.com/api_reference/client_side_api#currencies */
export type MoonpayCurrency = {
  id: string
  type: 'crypto' | 'fiat'
  name?: string
  code: string
  supportsTestMode: boolean
  supportsLiveMode: boolean
  notAllowedUSStates: string[]
  isSupportedInUS: boolean
  metadata?: {
    contractAddress: string
    chainId: string
  }
}
