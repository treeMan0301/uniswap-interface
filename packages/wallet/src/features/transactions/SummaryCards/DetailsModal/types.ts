import {
  ConfirmedSwapTransactionInfo,
  ExactInputSwapTransactionInfo,
  ExactOutputSwapTransactionInfo,
  OnRampPurchaseInfo,
  OnRampTransferInfo,
  UnknownTransactionInfo,
  WrapTransactionInfo,
} from 'wallet/src/features/transactions/types'

export type SwapTypeTransactionInfo =
  | ExactInputSwapTransactionInfo
  | ExactOutputSwapTransactionInfo
  | ConfirmedSwapTransactionInfo

import {
  ApproveTransactionInfo,
  FiatPurchaseTransactionInfo,
  NFTApproveTransactionInfo,
  NFTMintTransactionInfo,
  NFTTradeTransactionInfo,
  ReceiveTokenTransactionInfo,
  SendTokenTransactionInfo,
  TransactionType,
  TransactionTypeInfo,
  WCConfirmInfo,
} from 'wallet/src/features/transactions/types'

export function isApproveTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is ApproveTransactionInfo {
  return typeInfo.type === TransactionType.Approve
}

export function isFiatPurchaseTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is FiatPurchaseTransactionInfo {
  return typeInfo.type === TransactionType.FiatPurchase
}

export function isOnRampPurchaseTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is OnRampPurchaseInfo {
  return typeInfo.type === TransactionType.OnRampPurchase
}

export function isOnRampTransferTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is OnRampTransferInfo {
  return typeInfo.type === TransactionType.OnRampTransfer
}

export function isNFTApproveTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is NFTApproveTransactionInfo {
  return typeInfo.type === TransactionType.NFTApprove
}

export function isNFTMintTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is NFTMintTransactionInfo {
  return typeInfo.type === TransactionType.NFTMint
}

export function isNFTTradeTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is NFTTradeTransactionInfo {
  return typeInfo.type === TransactionType.NFTTrade
}

export function isReceiveTokenTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is ReceiveTokenTransactionInfo {
  return typeInfo.type === TransactionType.Receive
}

export function isSendTokenTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is SendTokenTransactionInfo {
  return typeInfo.type === TransactionType.Send
}

export function isSwapTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is SwapTypeTransactionInfo {
  return typeInfo.type === TransactionType.Swap
}

export function isWCConfirmTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is WCConfirmInfo {
  return typeInfo.type === TransactionType.WCConfirm
}

export function isWrapTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is WrapTransactionInfo {
  return typeInfo.type === TransactionType.Wrap
}

export function isUnknownTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is UnknownTransactionInfo {
  return typeInfo.type === TransactionType.Unknown
}
