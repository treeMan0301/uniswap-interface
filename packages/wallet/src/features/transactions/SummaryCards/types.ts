import { AuthTrigger } from 'uniswap/src/features/auth/types'
import { TransactionDetails } from 'uniswap/src/features/transactions/types/transactionDetails'
import { TransactionState } from 'uniswap/src/features/transactions/types/transactionState'
import { WalletChainId } from 'uniswap/src/types/chains'

export interface TransactionSummaryLayoutProps {
  authTrigger?: AuthTrigger
  transaction: TransactionDetails
  title?: string
  caption: string | JSX.Element
  icon?: JSX.Element
  index?: number
  onRetry?: () => void
}

export interface SummaryItemProps {
  authTrigger?: AuthTrigger
  transaction: TransactionDetails
  swapCallbacks?: SwapSummaryCallbacks
  index?: number
}

export interface SwapSummaryCallbacks {
  useLatestSwapTransaction: (address: string) => TransactionDetails | undefined
  useSwapFormTransactionState: (
    address: Address | undefined,
    chainId: WalletChainId | undefined,
    txId: string | undefined,
  ) => TransactionState | undefined
  onRetryGenerator?: (swapFormState: TransactionState | undefined) => () => void
}
