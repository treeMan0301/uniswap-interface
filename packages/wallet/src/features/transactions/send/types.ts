import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
import { AssetType, NFTAssetType } from 'uniswap/src/entities/assets'
import { WalletChainId } from 'uniswap/src/types/chains'
import { Account } from 'wallet/src/features/wallet/accounts/types'

interface BaseSendParams {
  type: AssetType
  txId?: string
  account: Account
  chainId: WalletChainId
  toAddress: Address
  tokenAddress: Address
  currencyAmountUSD?: Maybe<CurrencyAmount<Currency>> // for analytics
}

export interface SendCurrencyParams extends BaseSendParams {
  type: AssetType.Currency
  amountInWei: string
}

export interface SendNFTParams extends BaseSendParams {
  type: NFTAssetType
  tokenId: string
}

export type SendTokenParams = SendCurrencyParams | SendNFTParams
