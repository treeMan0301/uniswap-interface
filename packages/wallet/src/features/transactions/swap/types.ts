import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
import { CurrencyInfo } from 'uniswap/src/features/dataApi/types'
import { CurrencyField } from 'uniswap/src/features/transactions/transactionState/types'
import { WalletChainId } from 'uniswap/src/types/chains'
import { useTrade } from 'wallet/src/features/transactions/swap/trade/api/hooks/useTrade'
import { BaseDerivedInfo } from 'wallet/src/features/transactions/transfer/types'
import { WrapType } from 'wallet/src/features/transactions/types'

export type DerivedSwapInfo<
  TInput = CurrencyInfo,
  TOutput extends CurrencyInfo = CurrencyInfo,
> = BaseDerivedInfo<TInput> & {
  chainId: WalletChainId
  currencies: BaseDerivedInfo<TInput>['currencies'] & {
    [CurrencyField.OUTPUT]: Maybe<TOutput>
  }
  currencyAmounts: BaseDerivedInfo<TInput>['currencyAmounts'] & {
    [CurrencyField.OUTPUT]: Maybe<CurrencyAmount<Currency>>
  }
  currencyAmountsUSDValue: {
    [CurrencyField.INPUT]: Maybe<CurrencyAmount<Currency>>
    [CurrencyField.OUTPUT]: Maybe<CurrencyAmount<Currency>>
  }
  currencyBalances: BaseDerivedInfo<TInput>['currencyBalances'] & {
    [CurrencyField.OUTPUT]: Maybe<CurrencyAmount<Currency>>
  }
  focusOnCurrencyField: CurrencyField | null
  trade: ReturnType<typeof useTrade>
  wrapType: WrapType
  selectingCurrencyField?: CurrencyField
  txId?: string
  autoSlippageTolerance?: number
  customSlippageTolerance?: number
}
