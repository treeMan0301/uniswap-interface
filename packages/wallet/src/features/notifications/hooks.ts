import { useMemo } from 'react'
import { useSelector } from 'react-redux'
import { makeSelectHasNotifications } from 'wallet/src/features/notifications/selectors'
import { RootState } from 'wallet/src/state'

export function useSelectAddressHasNotifications(address: Address | null): boolean | undefined {
  const selectHasNotifications = useMemo(makeSelectHasNotifications, [])
  return useSelector((state: RootState) => selectHasNotifications(state, address))
}
