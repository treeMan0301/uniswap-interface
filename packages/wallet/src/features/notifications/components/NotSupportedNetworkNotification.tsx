import { useTranslation } from 'react-i18next'
import { AlertTriangle } from 'ui/src/components/icons'
import { iconSizes } from 'ui/src/theme'
import { NotificationToast } from 'wallet/src/features/notifications/components/NotificationToast'
import { NotSupportedNetworkNotification as NotSupportedNetworkNotificationType } from 'wallet/src/features/notifications/types'

export function NotSupportedNetworkNotification({
  notification: { hideDelay = 2000 },
}: {
  notification: NotSupportedNetworkNotificationType
}): JSX.Element | null {
  const { t } = useTranslation()

  return (
    <NotificationToast
      hideDelay={hideDelay}
      icon={<AlertTriangle color="$neutral2" size={iconSizes.icon36} />}
      title={t('extension.network.notSupported')}
    />
  )
}
