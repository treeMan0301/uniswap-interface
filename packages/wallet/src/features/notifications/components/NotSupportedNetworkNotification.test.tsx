import { AppNotificationType } from 'wallet/src/features/notifications/types'
import { renderWithProviders } from 'wallet/src/test/render'
import { NotSupportedNetworkNotification } from './NotSupportedNetworkNotification'

describe(NotSupportedNetworkNotification, () => {
  it('renders without error', () => {
    const tree = renderWithProviders(
      <NotSupportedNetworkNotification
        notification={{ type: AppNotificationType.NotSupportedNetwork }}
      />
    )

    expect(tree).toMatchSnapshot()
  })
})
