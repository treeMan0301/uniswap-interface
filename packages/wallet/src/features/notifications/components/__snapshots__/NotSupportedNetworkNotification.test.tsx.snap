// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotSupportedNetworkNotification renders without error 1`] = `
<View
  collapsable={false}
  forwardedRef={[Function]}
  handlerTag={1}
  handlerType="FlingGestureHandler"
  jestAnimatedStyle={
    {
      "value": {
        "transform": [
          {
            "translateY": -150,
          },
        ],
      },
    }
  }
  onGestureHandlerEvent={[Function]}
  pointerEvents="box-none"
  style={
    {
      "alignItems": "center",
      "borderBottomLeftRadius": 24,
      "borderBottomRightRadius": 24,
      "borderTopLeftRadius": 24,
      "borderTopRightRadius": 24,
      "flexDirection": "column",
      "justifyContent": "center",
      "position": "absolute",
      "top": 0,
      "transform": [
        {
          "translateY": -150,
        },
      ],
      "width": "100%",
      "zIndex": 1060,
    }
  }
>
  <View
    style={
      {
        "borderBottomColor": "#2222220D",
        "borderBottomLeftRadius": 24,
        "borderBottomRightRadius": 24,
        "borderBottomWidth": 1,
        "borderLeftColor": "#2222220D",
        "borderLeftWidth": 1,
        "borderRightColor": "#2222220D",
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopColor": "#2222220D",
        "borderTopLeftRadius": 24,
        "borderTopRightRadius": 24,
        "borderTopWidth": 1,
        "flexDirection": "column",
        "marginLeft": 24,
        "marginRight": 24,
        "shadowColor": "rgb(0,0,0)",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.07,
        "shadowRadius": 8,
      }
    }
  >
    <View
      cancelable={true}
      disabled={false}
      focusable={true}
      hitSlop={[Function]}
      minPressDuration={0}
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#F9F9F9",
          "borderBottomLeftRadius": 24,
          "borderBottomRightRadius": 24,
          "borderTopLeftRadius": 24,
          "borderTopRightRadius": 24,
          "flex": 1,
          "flexDirection": "row",
          "minHeight": 64,
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexBasis": "100%",
              "flexDirection": "row",
              "flexShrink": 1,
              "gap": 12,
              "justifyContent": "flex-start",
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="36"
            bbWidth="36"
            fill="currentColor"
            focusable={false}
            meetOrSlice={0}
            minX={0}
            minY={0}
            strokeWidth={8}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "color": "#7D7D7D",
                  "height": 36,
                  "width": 36,
                },
                {
                  "flex": 0,
                  "height": 36,
                  "width": 36,
                },
              ]
            }
            tintColor="#7D7D7D"
            vbHeight={24}
            vbWidth={24}
          >
            <RNSVGGroup
              fill={
                {
                  "type": 2,
                }
              }
              propList={
                [
                  "fill",
                  "strokeWidth",
                ]
              }
              strokeWidth="8"
            >
              <RNSVGPath
                d="M21.6074 17.1522L15.0004 4.79599C13.7204 2.40199 10.2814 2.40199 9.00042 4.79599L2.39349 17.1522C1.21949 19.3482 2.81353 22.0001 5.30853 22.0001H18.6923C21.1863 22.0001 22.7814 19.3472 21.6074 17.1522ZM11.2504 10.0001C11.2504 9.58609 11.5864 9.25009 12.0004 9.25009C12.4144 9.25009 12.7504 9.58609 12.7504 10.0001V14.0001C12.7504 14.4141 12.4144 14.7501 12.0004 14.7501C11.5864 14.7501 11.2504 14.4141 11.2504 14.0001V10.0001ZM12.0204 18.0001C11.4684 18.0001 11.0153 17.5521 11.0153 17.0001C11.0153 16.4481 11.4584 16.0001 12.0104 16.0001H12.0204C12.5734 16.0001 13.0204 16.4481 13.0204 17.0001C13.0204 17.5521 12.5724 18.0001 12.0204 18.0001Z"
                fill={
                  {
                    "type": 2,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <View
            style={
              {
                "alignItems": "flex-start",
                "flexDirection": "column",
                "flexShrink": 1,
              }
            }
          >
            <Text
              adjustsFontSizeToFit={true}
              allowFontScaling={false}
              maxFontSizeMultiplier={1.4}
              numberOfLines={2}
              style={
                {
                  "color": "#222222",
                  "fontFamily": "Basel-Book",
                  "fontSize": 16,
                  "lineHeight": 24,
                }
              }
              suppressHighlighting={true}
            >
              This network is not yet supported. More coming soon!
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
